🪟 Microsoft Teams channel support

Robi9 opened the pull request:
https://github.com/rapidpro/rapidpro/issues/1661

norkans7 commented on temba/channels/types/teams/type.py:
```diff
@@ -0,0 +1,34 @@
+    A Teams bot channel
+    """
+
+    CONFIG_TEAMS_APPLICATION_ID = "appID"
```
These should use the same code style as other config variables so use the snake style instead of the Camel case

norkans7 commented on templates/channels/types/teams/claim.haml:
```diff
@@ -0,0 +1,8 @@
+-load i18n
+
+-block pre-form
+  -blocktrans trimmed with name=brand.name
```
Can you add more details here, and include link to step by step guide if any

norkans7 reviewed (changes requested):
Can you include screenshots on how the channel is added and the needed config how to get them?

Robi9 commented:
I added more information about how to configure and get the data for the channel and in addition I also added a config page with some information and the Request URL for callback. Below are the prints of the information and the links, as I defined in the claim and in the config page I will send the prints of them even with the instructions.
Links dos prints: 

- [Create Bot channel registration in Azure account](https://docs.microsoft.com/en-us/microsoftteams/platform/sbs-teams-conversation-bot?tutorial-step=2#:~:text=registration%20in%20Azure-,account,-Access%20the%20Azure)
- [Portal Azure](https://portal.azure.com/)

![Captura de Tela 2022-08-25 às 20 12 05](https://user-images.githubusercontent.com/30378788/186926821-714df285-13f4-4681-98a0-b70d35b01ef5.png)
![Captura de Tela 2022-08-25 às 20 11 09](https://user-images.githubusercontent.com/30378788/186926860-6b29548e-57e2-4535-9c10-948559b5f19f.png)

matmsa27 commented:
Hey, @rowanseymour @norkans7 any news on this PR?

rowanseymour commented:
Hi @matmsa27 the truth is the addition of Slack (merged) and Teams has got us wondering where we should draw the line about what does and doesn't belong in RP. Some of the corporate collaboration tools (e.g. Slack, Teams) can be made to work with flows via bots in a private messages.. but don't always feel like a great fit compared to more social oriented chat apps. And everytime we hit merge on any channel type we're adding technical debt (and in the case of teams new courier depedencies). We've also put quite a bit of work into making channels pluggable (at least on the RP side) so there's the option for deployments to run channel types outside of what we merge into core. 

By the next major release we'll give a more concrete criteria on what we merge into core vs what we let deployments run in their own forks etc.
